op-service/cliapp/lifecycle.go

```diff
@@ -29,36 +29,26 @@ type Lifecycle interface {
 // a shutdown when the Stop context is not expired.
 type LifecycleAction func(ctx *cli.Context, close context.CancelCauseFunc) (Lifecycle, error)
 
-var interruptErr = errors.New("interrupt signal")
-
 // LifecycleCmd turns a LifecycleAction into an CLI action,
-// by instrumenting it with CLI context and signal based termination.
-// The signals are caught with the opio.BlockFn attached to the context, if any.
-// If no block function is provided, it adds default interrupt handling.
+// by instrumenting it with CLI context and signal based cancellation.
+// The signals are caught with the ctxinterrupt.waiter attached to the context, or default
+// interrupt signal handling if not already provided.
 // The app may continue to run post-processing until fully shutting down.
 // The user can force an early shut-down during post-processing by sending a second interruption signal.
 func LifecycleCmd(fn LifecycleAction) cli.ActionFunc {
 	return func(ctx *cli.Context) error {
-		hostCtx := ctx.Context
```

Comment:
The context flow here was all a bit janky, and affects all services

Reply:
The changes here make sense to me, nice work

Reply:
Sweet, thanks